File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        vector<int> freq(128, 0);

        for (char c : s) {
            freq[c]++;
        }

        int result = 0;
        bool hasOdd = false;

        for (int count : freq) {
            result += (count / 2) * 2;
            if (count % 2 == 1) {
                hasOdd = true;
            }
        }

        return hasOdd ? result + 1 : result;
    }
};

File: 0409-longest-palindrome/0409-longest-palindrome.go
func longestPalindrome(s string) int {
	freq := make([]int, 128)

	for _, ch := range s {
		freq[ch]++
	}

	result := 0
	hasOdd := false

	for _, count := range freq {
		result += (count / 2) * 2

		if count%2 == 1 {
			hasOdd = true
		}
	}

	if hasOdd {
		result++
	}

	return result
}